docs/API.md

```diff
@@ -1277,44 +1277,58 @@ Tags tags = minioClient.getObjectTags(
 |:---------------|:---------|:---------------------------|
 | ``bucketName`` | _String_ | Name of the bucket.        |
 | ``objectName`` | _String_ | Object name in the bucket. |
 
+
 | Returns                |
 |:-----------------------|
 | _String_ - URL string. |
 
  __Example__
  ```java
-String url = minioClient.getObjectUrl("my-bucketname", "my-objectname");
+// Get URL of an object useful when this object has public read access.
+String url =
+    minioClient.getObjectUrl("my-bucketname","my-objectname");
 System.out.println("my-bucketname/my-objectname can be downloaded by " + url);
 ```
 
  <a name="getPresignedObjectUrl"></a>
-### getPresignedObjectUrl(Method method, String bucketName, String objectName, Integer expires, Map<String,String> reqParams)
-`public String getPresignedObjectUrl(Method method, String bucketName, String objectName, Integer expires, Map<String,String> reqParams)` _[[Javadoc]](http://minio.github.io/minio-java/io/minio/MinioClient.html#getPresignedObjectUrl-io.minio.http.Method-java.lang.String-java.lang.String-java.lang.Integer-java.util.Map-)_
+### getPresignedObjectUrl(GetPresignedObjectUrlArgs args)
+`public String getPresignedObjectUrl(GetPresignedObjectUrlArgs args)` _[[Javadoc]](http://minio.github.io/minio-java/io/minio/MinioClient.html#getPresignedObjectUrl-io.minio.GetPresignedObjectUrlArgs-)_
 
 Gets presigned URL of an object for HTTP method, expiry time and custom request parameters.
 
  __Parameters__
-| Parameter      | Type                 | Description                            |
-|:---------------|:---------------------|:---------------------------------------|
-| ``method``     | _[Method]_           | HTTP method to generate presigned URL. |
-| ``bucketName`` | _String_             | Name of the bucket.                    |
-| ``objectName`` | _String_             | Object name in the bucket.             |
-| ``expires``    | _Integer_            | Expiry in seconds; defaults to 7 days. |
-| ``reqParams``  | _Map<String,String>_ | Request parameters to override.        |
+ 
+| Parameter   | Type                           | Description  |
+|:------------|:-------------------------------|:-------------|
+| ``args``    | _[GetPresignedObjectUrlArgs]_  | Arguments.   |
+
 
 | Returns                |
 |:-----------------------|
 | _String_ - URL string. |
 
  __Example__
  ```java
-String url = minioClient.getPresignedObjectUrl(Method.DELETE, "my-bucketname", "my-objectname", 24 * 60 * 60,
-    reqParams);
+// Get presigned URL of an object for HTTP method, expiry time and custom request parameters.
+Map<String, String> reqParams = new HashMap<>();
+      reqParams.put("response-content-type", "application/json");
+      
+String url =
+    minioClient.getPresignedObjectUrl(
+        GetPresignedObjectUrlArgs.builder()
+            .method(Method.DELETE)
+            .bucket("my-bucketname")
+            .object("my-objectname")
+            .expires(24 * 60 * 60)
```

Comment:
I think we can have one more version of this method to accept `(int duration, TimeUnit unit)`. I don't know `TimeUnit` supports days. We could find whichever class supports wider range.

Reply:
TimeUnit supports Days. Incorporated the change